bbq/vm/context.go

```diff
@@ -55,6 +57,8 @@ type Context struct {
 	// TODO: Maybe extend/share this between executions.
 	semaTypes map[sema.TypeID]sema.Type
 
+	attachmentIterationMap map[*interpreter.CompositeValue]bool
```

Comment:
Small improvements: Maybe just store an empty `struct{}` as the value and use existence. See suggestions below

```suggestion
	attachmentIterationMap map[*interpreter.CompositeValue]struct{}
```